consult-org.el

```diff
@@ -0,0 +1,167 @@
+       (setq opoint (point))
+       (pcase-let ((`(_ ,level ,todo ,prio) (org-heading-components)))
+         (consult--location-candidate
+          (org-format-outline-path (org-get-outline-path t t))
+          (point-marker)
+          line
+          'consult-org--level level
+          'consult-org--todo todo
+          'consult-org--priority prio)))
+     match scope skip)))
+
+;;;###autoload
+(defun consult-org-heading (&optional match scope)
+  "Jump to an Org heading.
+
+MATCH and SCOPE are as in `org-map-entries' and determine which
+entries are offered.  By default, all entries of the current
+buffer are offered."
+  (interactive (unless (derived-mode-p 'org-mode)
+                 (user-error "Must be called from an Org buffer.")))
+  (consult--jump
+   (consult--read
+    (consult--with-increased-gc (consult-org--entries match scope))
+    :prompt "Go to heading: "
+    :category 'consult-org-heading
+    :sort nil
+    :title (unless (member scope '(nil 'tree 'region 'region-start-level 'file))
+             ;; Don't add titles when only showing entries from current buffer
+             (lambda (cand)
+               (let ((marker (car (get-text-property 0 'consult-location cand))))
+                 (buffer-name (marker-buffer marker)))))
+    :narrow (consult-org--narrow)
+    :require-match t
+    :lookup #'consult--lookup-location
+    :history '(:input consult-org--history)
+    :state (consult--jump-preview))))
+
+;;;###autoload
+(defun consult-org-agenda (&optional match)
+  "Jump to an Org agenda heading.
+
+By default, all agenda entries are offered.  MATCH is as in
+`org-map-entries' and can used to refine this."
+  (interactive)
+  (unless org-agenda-files
+    (user-error "No agenda files"))
+  (consult-org-heading match 'agenda))
+
+;;;###autoload
+(defun consult-org-clock (&optional match scope)
+  "Clock into an Org heading.
+
+MATCH and SCOPE are as in `org-map-entries' and determine which
+entries are offered.  By default, offer entries of files with a
+recent clocked item."
+  (interactive)
+  (setq scope (or scope
+                  (thread-last (progn (org-clock-load) org-clock-history)
+                    (mapcar 'marker-buffer)
+                    (mapcar 'buffer-file-name)
+                    seq-uniq
+                    (seq-remove 'null))
+                  (user-error "No recent clocked tasks")))
+  (org-clock-clock-in
+   (list
+    (consult--read
+     (consult--with-increased-gc
+      (sort ;; Recent items first, everything else in their original order
+       (consult-org--entries match scope)
+       (lambda (c1 c2)
+         (let ((m1 (car (get-text-property 0 'consult-location c1)))
+               (m2 (car (get-text-property 0 'consult-location c2))))
+           (and (member m1 org-clock-history)
+                (not (member m1 (member m2 org-clock-history))))))))
+     :prompt "Clock in: "
+     :category 'consult-org-heading
```

Comment:
Why did you decide for an extra category now? It may be better to use `consult-location` because this is what I used now for general location strings. However as I mentioned befre, if you want to attach org specific actions, to headings a separate category is better (maybe even 'org-heading). But as of now there are no embark actions for org headings and I am unsure if they are worth it since org already has its own set of commands to manipulate the headings. cc @oantolin

Reply:
See also https://github.com/oantolin/embark/issues/95#issuecomment-758246901 by  @sheijk

Reply:
I think you are right, this would be too fine grained. This would be fine (but also maybe precious) if we could declare that `consult-org-heading` inherits from `consult-location`. That not being the case, I'll revert to `consult-location`. And I agree that some of the editing features discussed in the linked comment are already covered satisfactorily in Org itself.

Reply:
> This would be fine (but also maybe precious) if we could declare that consult-org-heading inherits from consult-location. That not being the case, I'll revert to consult-location.

You can actually do that in Embark, a `consult-org-heading` keymap could inherit from the `consult-location` keymap. But I think using a separate category (Embark target type) for org headings would only make sense if Embark would support actions at point for org headings (then the category 'org-heading would make more sense, since it could also work without consult).